### PR TITLE
exp/txnbuild: change trust

### PR DESCRIPTION
**Pull request by ire-and-curses**
As part of #982, adds change trust operation to `txnbuild`. Comparing to branch since non-native assets are not merged yet.

### CODE REVIEW

File: exp/txnbuild/change_trust.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// BuildXDR for ChangeTrust returns a fully configured XDR Operation.
+func (ct *ChangeTrust) BuildXDR() (xdr.Operation, error) {
+	var err error
+	if ct.Line.IsNative() {
+		return xdr.Operation{}, errors.New("Trustline cannot be extended to a native (XLM) asset")
+	}
+	ct.xdrOp.Line, err = ct.Line.ToXDR()
+	if err != nil {
+		return xdr.Operation{}, errors.Wrap(err, "Can't convert trustline to XDR")
```
Comment (poliha) [inline exp/txnbuild/change_trust.go]: Maybe change to  `Can't convert asset for trustline to XDR` to indicate here that its the asset conversion that is failing..

File: exp/txnbuild/asset.go [hunk 2]
```diff
@@ -26,12 +26,20 @@ func NewAsset(code, issuer string) *Asset {
...
 }
 
+// IsNative for Asset returns true if this is an XLM asset.
+func (a *Asset) IsNative() bool {
+	// Native (Lumens) has no code or issuer
+	if a.Code == "" && a.Issuer == "" {
+		return true
+	}
+	return false
+}
```
Comment (bartekn) [inline exp/txnbuild/asset.go]: Can be changed to one-liner:
```go
return a.Code == "" && a.Issuer == ""
```

File: exp/txnbuild/change_trust.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"github.com/stellar/go/amount"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/xdr"
+)
+
+// ChangeTrust represents the Stellar change trust operation. See
+// https://www.stellar.org/developers/guides/concepts/list-of-operations.html
+type ChangeTrust struct {
+	Line  Asset
```
Comment (bartekn) [inline exp/txnbuild/change_trust.go]: I'm wondering if we should change this to `*Asset` too, for consistency.

File: exp/txnbuild/change_trust.go (hunk 1 above)
Comment (ire-and-curses) [reply exp/txnbuild/change_trust.go]: :+1:

File: exp/txnbuild/asset.go (hunk 2 above)
Comment (ire-and-curses) [reply exp/txnbuild/asset.go]: :+1:

File: exp/txnbuild/change_trust.go (hunk 3 above)
Comment (ire-and-curses) [reply exp/txnbuild/change_trust.go]: Yeah I went back and forth about that. I'll change it.

**Pull request merged**